modules/okhttp-client/src/main/java/com/spotify/apollo/http/client/LoggingHttpClient.java

```diff
@@ -0,0 +1,107 @@
+/*-
+ * -\-\-
+ * Spotify Apollo okhttp Client Module
+ * --
+ * Copyright (C) 2013 - 2021 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.apollo.http.client;
+
+import com.spotify.apollo.Request;
+import com.spotify.apollo.Response;
+import com.spotify.apollo.environment.IncomingRequestAwareClient;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricRegistry;
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nullable;
+import okio.ByteString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggingHttpClient implements IncomingRequestAwareClient {
+
+  private static final Logger logger = LoggerFactory.getLogger(LoggingHttpClient.class);
+
+  private final IncomingRequestAwareClient client;
+  private final SemanticMetricRegistry metricRegistry;
+  private final MetricId baseId = MetricId.build();
+
+  public LoggingHttpClient(
+      IncomingRequestAwareClient client, SemanticMetricRegistry metricRegistry) {
+    this.client = client;
+    this.metricRegistry = metricRegistry;
+  }
+
+  @Override
+  public CompletionStage<Response<ByteString>> send(Request request, Optional<Request> incoming) {
+
+    if (!request.uri().startsWith("http")) {
+      return client.send(request, incoming);
+    }
+
+    final long start = System.currentTimeMillis();
+
+    return client
+        .send(request, incoming)
+        .whenComplete(
+            (response, throwable) -> {
+              final long elapsedMillis = System.currentTimeMillis() - start;
+              if (throwable != null) {
+                logOutgoingRequest(request, elapsedMillis, null);
+                logger.error("Exception caught: ", throwable);
```

Comment:
typically when you have a client class like this which wraps another instance, you shouldn't log exceptions that will be logged by the code using this also - I think logging this here means its likely a single exception will be logged multiple times.